### DISCUSSION TITLE
Definition of Done: Design

### DISCUSSION THREAD

**Discussion by eliasfaltin** [GECO-Project-App/facilitation-app#51] [How we work]
I propose the following definition of done for design work on the project (signaling the design is ready to be handed over to developers):

- [ ] the design is supported by the [Design Pillars](https://github.com/GECO-Project-App/facilitation-app/wiki/Design-Pillars) 
- [ ] wherever possible and reasonable, components and previously implemented elements are re-used
- [ ] changes to existing components are clearly labeled on the component page, and an additional note about component changes is present on the main page in Figma
- [ ] implemented design into the main page in Figma (mapped out screen flow and high-fidelity mock-ups for each screen)
- [ ] all graphics that are to be replaced with a Rive animation are marked
- [ ] a rive file with (at least) a static, placeholder image is submitted to the repository
- [ ] All required inputs for a given Rive animation are documented on the main page in Figma

<details><summary>Comments</summary>
<p>

- I don't think Rive animations need to be final yet for developers to start working on the screen in question. But inputs need to be defined.
- the placeholder image should be ugly so we always remember to replace it with an animation. Alternatively, we could just put a "still frame" into the rive file so the placeholder looks nicer. But those tend to stay in games/apps because they don't look shit enough to be worth the effort to replace 😅

</p>
</details>

**Comment by aripuguh** [comment]
Noted @eliasfaltin ! The only comment I have is about the placeholder image for Rive animation, I am not sure whether using an ugly image would make them remember, but rather potentially confuse them, on which Rive animation should go in that section. 😅 My suggestion is to add a red frame/vibrant color frame that indicates that they should replace it with a Rive document (see the reference attached). LMK what you think!
<img width="450" alt="Screen Shot 2024-09-17 at 6 55 58 PM" src="https://github.com/user-attachments/assets/d69b6b37-83db-4b31-a289-e3e61b864a42">

**Reply by eliasfaltin** [reply]
That sounds reasonable. There are countless stories from the game industry where people put in placeholder art that never gets replaced (Starcraft II load screens, for example). So, the idea behind making it intentionally ugly is that there is no way we would want the ugliness to be seen by the world.

But if we go in this direction, it might be best to just put in a "static" rive file from the beginning. That way, it is easier for you to update things independently of @alice-ang and @Cristian-Nass .

**Reply by aripuguh** [reply]
Ah okej! That made sense @eliasfaltin ... Well then it's up to @alice-ang & @Cristian-Nass on which way they are comfortable with so they don't miss the Rive animation part 👻. I can adjust.

**Reply by alice-ang** [reply]
Hmm yeah a static rive file (can be just a red box) in the repo might be good. Just so we have some asset to link to when we're implementing the code. And Ari can finish the motion graphics with no stress of blocking us.


Regarding how the graphics in the design that are to be replaced by rive should be marked, it doesn't matter to me. The dashed red border is fine for me. If we're going with having a static rive file in the repo maybe we can just add the filename somewhere to make it more clear.
<img width="467" alt="Screenshot 2024-09-17 at 18 01 32" src="https://github.com/user-attachments/assets/fe08cd05-b87a-4616-a18b-007a123a1625">

**Reply by alice-ang** [reply]
And if we create a pull request template we can also put in reminders for us to double check that all graphics have been replaced with the rive animations when we open prs 😄
